gateway/assets/script/bootstrap.js

```diff
@@ -169,6 +165,14 @@ app.controller("home", ['$scope', '$log', '$http', '$location', '$timeout', '$md
                     if (response && response.data) {
                         if (previousItems.length != response.data.length) {
                             $scope.functions = response.data;
+                            
+                            // update the selected function object because the newly fetched object from the API becomes a different object
+                            var filteredSelectedFunction = $filter('filter')($scope.functions, {name: $scope.selectedFunction.name}, true);
+                            if (filteredSelectedFunction && filteredSelectedFunction.length > 0) {
+                                $scope.selectedFunction = filteredSelectedFunction[0];
+                            } else {
+                                $scope.selectedFunction = undefined;
+                            }
```

Comment:
Lines above is the main part of the fix. Where it updates the selectedFunction to the new object.

Reply:
What does `$filter` do? is it the same effect as a for loop and matching on name?

Reply:
Yes. I wanted to use [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) but it's a relatively new function so didn't want to break older browsers.

I wondered if I should add a simple `find` function and call that, or to use `$filter` but chose to take advantage of AngularJS's built in function.